{
  try {
    validation.executeMethod(method,client,protocol);
  }
 catch (  final HttpException e) {
    s_logger.error("HttpException caught while trying to connect to the REST Service",e);
    method.releaseConnection();
    throw new CloudstackRESTException("API call to REST Service Failed",e);
  }
catch (  final IOException e) {
    s_logger.error("IOException caught while trying to connect to the REST Service",e);
    method.releaseConnection();
    throw new CloudstackRESTException("API call to Nicira REST Service Failed",e);
  }
}
